exercises/list-ops/listops_test.go

```diff
@@ -0,0 +1,278 @@
+	list     IntList
+	fn       unaryFunc
+	want     IntList
+}{
+	{
+		name:     "empty list",
+		property: "map",
+		list:     []int{},
+		fn:       func(x int) int { return x + 1 },
+		want:     []int{},
+	},
+	{
+		name:     "non-empty list",
+		property: "map",
+		list:     []int{1, 3, 5, 7},
+		fn:       func(x int) int { return x + 1 },
+		want:     []int{2, 4, 6, 8},
+	},
+}
+
+func TestMapMethod(t *testing.T) {
+	for _, tt := range mapTestCases {
+		got := tt.list.Map(tt.fn)
+		if !reflect.DeepEqual(tt.want, got) {
+			t.Fatalf("Build for test case %q for property %s returned %v but was expected to return %v",
+				tt.name, tt.property, got, tt.want)
+		}
+	}
+}
+
+var reverseTestCases = []struct {
+	name     string
+	property string
+	list     IntList
+	want     IntList
+}{
+	{
+		name:     "empty list",
+		property: "reverse",
+		list:     []int{},
+		want:     []int{},
+	},
+	{
+		name:     "non-empty list",
+		property: "reverse",
+		list:     []int{1, 3, 5, 7},
+		want:     []int{7, 5, 3, 1},
+	},
+}
+
+func TestReverseMethod(t *testing.T) {
+	for _, tt := range reverseTestCases {
+		got := tt.list.Reverse()
+		if !reflect.DeepEqual(tt.want, got) {
+			t.Fatalf("Build for test case %q for property %s returned %v but was expected to return %v",
+				tt.name, tt.property, got, tt.want)
+		}
+	}
+}
+
+var appendTestCases = []struct {
+	name       string
+	property   string
+	list       IntList
+	appendThis IntList
+	want       IntList
+}{
+	{
+		name:       "empty list",
+		property:   "append",
+		list:       []int{},
+		appendThis: []int{},
+		want:       []int{},
+	},
+	{
+		name:       "empty list to list",
+		property:   "append",
+		list:       []int{},
+		appendThis: []int{1, 2, 3, 4},
+		want:       []int{1, 2, 3, 4},
+	},
+	{
+		name:       "non-empty lists",
+		property:   "append",
+		list:       []int{1, 2},
+		appendThis: []int{2, 3, 4, 5},
+		want:       []int{1, 2, 2, 3, 4, 5},
+	},
+}
+
+func TestAppendMethod(t *testing.T) {
+	for _, tt := range appendTestCases {
+		got := tt.list.Append(tt.appendThis)
+		if !reflect.DeepEqual(tt.want, got) {
+			t.Fatalf("Build for test case %q for property %s returned %v but was expected to return %v",
+				tt.name, tt.property, got, tt.want)
+		}
+	}
+}
+
+var concatTestCases = []struct {
+	name     string
+	property string
+	list     IntList
+	args     []IntList
+	want     IntList
+}{
+	{
+		name:     "empty list",
+		property: "append",
+		list:     []int{},
+		args:     []IntList{},
+		want:     []int{},
+	},
+	{
+		name:     "list of lists",
+		property: "append",
+		list:     []int{1, 2},
+		args:     []IntList{[]int{3}, []int{}, []int{4, 5, 6}},
+		want:     []int{1, 2, 3, 4, 5, 6},
+	},
+}
+
+func TestConcatMethod(t *testing.T) {
+	for _, tt := range concatTestCases {
+		got := tt.list.Concat(tt.args)
+		if !reflect.DeepEqual(tt.want, got) {
+			t.Fatalf("Build for test case %q for property %s returned %v but was expected to return %v",
+				tt.name, tt.property, got, tt.want)
```

Comment:
Consider using something along the lines of:
```
  t.Fatalf("FAIL: %q ... ", tt.name, ... )
```
Also, for brevity,  could use "expected: xxx  actual: yyy" pattern if you want. Survey some of the other exercises test code output for reference.